panel/param.py

```diff
@@ -227,7 +227,8 @@ def _update_widgets(self, *events):
                 parameters = [] if event.new == [] else event.new
 
         if parameters != [] and parameters != self.parameters:
-            self.parameters = parameters
+            if not self.parameters:
+                self.parameters = parameters
```

Comment:
This change does not seem correct to me, but I can see how the previous logic was incorrect too. Will keep your test though.

Reply:
makes sense